[Chen-Kuei] iP

Chen-Kuei opened the pull request:
# Hi! I'm BiuBiu :D
> Biu!

### BiuBiu frees your mind of having to remember things you need to do. It's,

1. text-based
1. easy to learn
1. ~~FAST SUPER~~ **FAST** to use

### All you need to do is,

- download it from [HERE](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
- double-click on it.
- add your tasks.
- let it manage your tasks for you :smiling_face_with_three_hearts:

## And it is **FREE**!

### Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

seanleong339 committed:
.gitignore: Fix ACTUAL.txt -> ACTUAL.TXT  (#73)

text-ui-test/runtest.bat|sh scripts generate a file ACTUAL.TXT.
However, .gitignore uses ACTUAL.txt, which means the generated
file will not be ignored by Git on non-Windows OS.

Let's update .gitignore as ACTUAL.txt -> ACTUAL.TXT

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Echo

Commit:
Add, List

Commit:
Mark As Done

Commit:
Revert "Mark As Done"

This reverts commit d8a0c189ca52ffcf8dcbb34b117bd947b5cf08b4.

Commit:
Mark As Done

Commit:
ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Revert "Automated Text UI Testing"

This reverts commit 59487b8d91bb9e6b12ff379adec6b7bc40fbab0e.

Commit:
Revert "Revert "Automated Text UI Testing""

This reverts commit 965afecee03111ac72567184e4a3182a9588e7da.

Commit:
Handle Errors

Commit:
Merge commit '6754850f9a250ef34efd3d89a8900a789b268630'

Commit:
Delete

Commit:
Save

Commit:
Merge branch 'branch-Level-7'

Commit:
Dates and Times

Commit:
Merge branch 'branch-Level-8'

Commit:
OOP(Ui, TaskList, Parser, Storage)

Commit:
Merge branch 'master' into add-gradle-support

Commit:
Add JUnit tests

Commit:
Merge branch 'add-gradle-support'

Commit:
Divide classes into packages

Commit:
Package the App as a JAR file

Commit:
Add JavaDoc comments

Commit:
Coding standard

Commit:
Revert "Coding standard"

This reverts commit 83e9a4424bd23d8bb237dc191ededb1f11591fe0.

Commit:
Coding Standard

zacwong2151 commented on docs/duke.txt:
```diff
@@ -0,0 +1,3 @@
+TODO | read book
+DEADLINE | project | 2022-10-10
+EVENT | meet | 2023-06-12 1400 | 2023-06-12 1600
```
correct me if I'm wrong, but I believe that this file should be added to your .gitignore file

Commit:
Find

Commit:
Merge branch 'branch-Level-9' into branch-A-CodingStandard

# Conflicts:
#	src/main/java/duke/Ui.java

Commit:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Event.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Task.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Todo.java
#	src/main/java/duke/Ui.java

zacwong2151 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,157 @@
+                        Storage.TaskType taskType = Storage.TaskType.valueOf(taskData[0]);
+                        String taskDescription = taskData[1];
+                        String taskTime1 = (taskData.length > 2) ? taskData[2] : "";
+                        String taskTime2 = (taskData.length > 3) ? taskData[3] : "";
```
not very sure what these two task times represent? some commented code would be appreciated

zacwong2151 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,157 @@
+                    storage.saveTasks(tasks);
+                    break;
+                case "unmark":
+                    int notDoneTaskIndex = Integer.parseInt(parts[1]) - 1;
```
great job on writing a clear variable name!

Commit:
Merge all branches

zacwong2151 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,53 @@
+public class Event extends Task {
+    // protected String from;
+    // protected String to;
+    private LocalDateTime startTime;
+    private LocalDateTime endTime;
```
nice to see that you went beyond what was asked in the iP and made Event have LocalDateTime fields as well. However, this causes some user experience problems, because now in order to enter a Event task you have to write a whole big chunk of text 😬

zacwong2151 reviewed (commented):
LGTM! Here are a few recommendations
1. add javadocs!
2. idk, great job on this!

hcs1203 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,283 @@
+//            saveTask();
+//        }
+//    }
+//}
```
Remove commented code. As long as this is a previous version, it will be saved in version control, or store in a separate txt file.

hcs1203 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,283 @@
+TODO | read book
+DEADLINE | project | 2022-10-10
+EVENT | meeting | 2023-06-12 1400 | 2023-06-12 1800
+ */
```
State todo on top, delete when done

hcs1203 reviewed (commented):
Well done! Good and clear code with accurate descriptions and good naming mechanisms. Can remove parts of code that make it look like a draft

Commit:
Add a GUI to the chatbot

Commit:
Merge branch 'branch-Level-10'

adammangzijun commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,48 @@
+        String[] commandParts = userCommand.split(" ", 2);
+        String taskType = commandParts[0].toLowerCase();
+
+        switch (taskType) {
```
Cases should be in the same indentation as the switch keyword.

adammangzijun commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,81 @@
+     * @return The parsed deadline date as a LocalDate object, or null if parsing fails.
+     */
+    private LocalDate parseDeadline(String deadlineString) {
+
```
I'm not sure if this is an accident but there should be consistency in whether you separate your block statements with 1 line spacing or without.

adammangzijun reviewed (commented):
Overall a well done and concise piece of code. Good job! 😄

Commit:
use checkstyle with gradle

Commit:
user assert feature

Commit:
Improve code quality

Removing redundant code in each java class and correcting the
indentation for 'case' clauses improve the code quality.

Let's remove the indentation of 'case' clauses.

Chen-Kuei committed:
Merge pull request #1 from Chen-Kuei/branch-assertions

use assert feature

Chen-Kuei committed:
Merge branch 'master' into branch-CodeQuality

Chen-Kuei committed:
Merge pull request #2 from Chen-Kuei/branch-CodeQuality

Improve code quality

Commit:
improve code quality2.0

Commit:
Merge branch 'branch-assertions' of https://github.com/Chen-Kuei/ip

Commit:
improve code quality3.0

Commit:
Merge branch 'master' of https://github.com/Chen-Kuei/ip

# Conflicts:
#	src/main/java/duke/Duke.java

Commit:
update GUI design and improve code quality

Commit:
update GUI

Commit:
C-Update

Commit:
Merge branch 'branch-C-Update'

Commit:
add a user guide

Commit:
update user guide

Commit:
add title bar of the window, update Ui.png

Commit:
update user guide, update main window

Commit:
improve code quality(modify too-long methods)

Commit:
Merge branch 'branch-A-CodeQuality'

Commit:
Correct the 'loadTasks' method, remove redundant code

Commit:
create a jar file

Commit:
Compile with Java 11 JDK
